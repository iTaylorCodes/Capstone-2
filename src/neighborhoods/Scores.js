import Score from "./Score";

const Scores = ({ scores }) => {
  return (
    <div className="Scores row p-1 d-flex justify-content-center">
      {scores.bikeScore && scores.bikeScore.score ? (
        <div className="col-4 col-md-3 col-lg-5">
          <Score score={scores.bikeScore} />
        </div>
      ) : (
        ""
      )}
      {scores.walkScore && scores.walkScore.score ? (
        <div className="col-4 col-md-3 col-lg-5">
          <Score score={scores.walkScore} />
        </div>
      ) : (
        ""
      )}
      {scores.transitScore && scores.transitScore.score ? (
        <div className="col-4 col-md-3 col-lg-5">
          <Score score={scores.transitScore} />
        </div>
      ) : (
        ""
      )}
    </div>
  );
};

export default Scores;
