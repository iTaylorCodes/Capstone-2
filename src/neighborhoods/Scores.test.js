import Scores from "./Scores";
import { render } from "@testing-library/react";

let scores = {
  walkScore: { score: 10, description: "score" },
  bikeScore: { score: 10, description: "score" },
  transitScore: { score: 10, description: "score" },
};

it("renders without crashing", () => {
  render(<Scores scores={scores} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Scores scores={scores} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders when scores are unavailable", () => {
  scores = {
    walkScore: { score: null, description: null },
    bikeScore: { score: null, description: null },
    transitScore: { score: null, description: null },
  };

  render(<Scores scores={scores} />);
});
