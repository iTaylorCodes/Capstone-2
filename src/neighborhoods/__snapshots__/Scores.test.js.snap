// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Scores row p-1 d-flex justify-content-center"
  >
    <div
      class="col-4 col-md-3 col-lg-5"
    >
      <div
        class="Score m-3 rounded-circle p-2"
        style="width: 125px; height: 125px; background: rgb(93, 217, 93);"
      >
        <h2>
          10
        </h2>
        <p>
          score
        </p>
      </div>
    </div>
    <div
      class="col-4 col-md-3 col-lg-5"
    >
      <div
        class="Score m-3 rounded-circle p-2"
        style="width: 125px; height: 125px; background: rgb(93, 217, 93);"
      >
        <h2>
          10
        </h2>
        <p>
          score
        </p>
      </div>
    </div>
    <div
      class="col-4 col-md-3 col-lg-5"
    >
      <div
        class="Score m-3 rounded-circle p-2"
        style="width: 125px; height: 125px; background: rgb(93, 217, 93);"
      >
        <h2>
          10
        </h2>
        <p>
          score
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
